es6/lib/createRouteHandler.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+// jshint -W040
+
+import Boom from 'boom';
+import camelCase from 'lodash/string/camelCase';
+import first from 'lodash/array/first';
+import forEach from 'lodash/collection/forEach';
+import isEmpty from 'lodash/lang/isEmpty';
+import mapKeys from 'lodash/object/mapKeys';
+import omit from 'lodash/object/omit';
+import rearg from 'lodash/function/rearg';
+import snakeCase from 'lodash/string/snakeCase';
+
+function apiController(options, knexClient) {
+  return {
+
+    create(request, reply) {
+      var resource;
+
+      if (options.tableHeaderFormat === 'snakeCase') {
+        var snakeCaseKeys = rearg(snakeCase, [1, 0]);
+        resource = mapKeys(request.payload, snakeCaseKeys);
+      } else {
+        resource = request.payload;
+      }
+
```

Comment:
what about passing in `keyInTransformFn` and `keyOutTransformFn` and it can default `_.identity` but you could pass in `_.snakeCase`, `_.camelCase`, etc ... ?